Google.PowerShell/BigQuery/BqTabledata.cs

```diff
@@ -0,0 +1,135 @@
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "BqSchema")]
+    public class NewBqSchema : PSCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The existing TableSchema that you wish to add a column to. 
+        /// If this value is not present, a new schema will be created.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        public TableSchema InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the column to be added. The name must be unique among columns in each schema.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The type of the column to be added. Possible values include STRING, BYTES, INTEGER (INT64), 
+        /// FLOAT (FLOAT64), BOOLEAN (BOOL), TIMESTAMP, DATE, TIME, DATETIME, and RECORD (STRUCT).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateSet("STRING", "BYTES", "INTEGER", "INT64", "FLOAT", "FLOAT64", "BOOLEAN", "BOOL", 
+            "TIMESTAMP", "DATE", "TIME", "DATETIME", "RECORD", "STRUCT")]
+        public string Type { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An optional description for this column.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The mode of the column to be added. Possible values include NULLABLE, REQUIRED, and REPEATED.
+        /// The default value is NULLABLE.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateSet("NULLABLE", "REQUIRED", "REPEATED", IgnoreCase = true)]
+        public string Mode { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Describes the optional nested schema fields if the type property is set to RECORD. Pass in 
+        /// another TableSchema object and this cmdlet will properly nest its fields in the new column.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public TableSchema Fields { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (InputObject == null)
+            {
+                InputObject = new TableSchema();
+                InputObject.Fields = new List<TableFieldSchema>();
+            }
+
+            if (Mode == null)
+            {
+                Mode = "NULLABLE";
+            }
+
+            foreach (TableFieldSchema t in InputObject.Fields)
```

Comment:
Yep.  As I was writing this I was thinking about a lambda expression kinda thing, but I wasn't confident in my C# foo.  

And about the errors, yes.  There's no point in continuing on after that, so it'll be better not to harm the original array.  I'll add something to the docs about that.